web/src/components/profileEdit/QuotaCardEdit.tsx

```diff
@@ -65,23 +80,67 @@ const QuotaCardEdit: React.FC<IQuotaCardEditProps> = (props) => {
 
   const [goBackToProfileEditable, setGoBackToProfileEditable] = useState<boolean>(false);
   const [specs, setSpecs] = useState<any>([]);
-  const [applyingQuotaSpecs, setApplyingQuotaSpecs] = useState<any>([]);
-  const [quotaSizes, setQuotaSizes] = useState<any>({});
+  const [applyingQuotaSpecs, setApplyingQuotaSpecs] = useState<any>({});
+  const [quotaSizes, setQuotaSizes] = useState<any>([]);
+
+  const getCorrespondingQuota = (selectedSize: ProjectResourceQuotaSize): QuotaSpecsInterface => {
+    let result: QuotaSpecsInterface = {
+      cpuNums: [],
+      memoryNums: [],
+      storageNums: [],
+    };
+    if (quotaSizes.length > 0) {
+      Object.keys(quotaSize).forEach((key) => {
+        switch (key) {
+          case 'quotaCpuSize':
+            result.cpuNums =
+              quotaSizes
+                .filter((size: any) => size.name.toUpperCase() === selectedSize[key].toUpperCase())
+                .pop()?.cpuNums || [];
```

Comment:
Billy.... what is wrong with all of these case statements? I will stop reviewing now.